python-phoenixdb/phoenixdb/avatica/client.py

```diff
@@ -149,6 +149,7 @@ def __init__(self, url, max_retries=None, auth=None, verify=None):
         self.auth = auth
         self.verify = verify
         self.connection = None
+        self.session = requests.Session()
```

Comment:
IMO this would better in open_connection()

Reply:
actually, I've pushed this down into `_post_request` to avoid having to create a bunch of "if not self.session` checks.